core/cosmosdb/cache-invalidator/src/main/scala/org/apache/openwhisk/core/database/cosmosdb/cache/WhiskChangeEventObserver.scala

```diff
@@ -19,30 +19,34 @@ package org.apache.openwhisk.core.database.cosmosdb.cache
 
 import akka.Done
 import akka.event.slf4j.SLF4JLogging
-import com.microsoft.azure.documentdb.Document
-import com.microsoft.azure.documentdb.changefeedprocessor.ChangeFeedObserverContext
+import com.azure.data.cosmos.CosmosItemProperties
+import com.azure.data.cosmos.internal.changefeed.ChangeFeedObserverContext
 import kamon.metric.MeasurementUnit
 import org.apache.openwhisk.common.{LogMarkerToken, MetricEmitter}
 import org.apache.openwhisk.core.database.CacheInvalidationMessage
 import org.apache.openwhisk.core.database.cosmosdb.CosmosDBConstants
-import org.apache.openwhisk.core.entity.CacheKey
 import org.apache.openwhisk.core.database.cosmosdb.CosmosDBUtil.unescapeId
+import org.apache.openwhisk.core.entity.CacheKey
 
 import scala.collection.concurrent.TrieMap
 import scala.collection.immutable.Seq
-import scala.concurrent.{Await, Future}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
 
-class WhiskChangeEventObserver(config: InvalidatorConfig, eventProducer: EventProducer) extends ChangeFeedObserver {
+class WhiskChangeEventObserver(config: InvalidatorConfig, eventProducer: EventProducer)(implicit ec: ExecutionContext)
+    extends ChangeFeedObserver {
   import WhiskChangeEventObserver._
 
-  override def process(context: ChangeFeedObserverContext, docs: Seq[Document]): Unit = {
+  override def process(context: ChangeFeedObserverContext, docs: Seq[CosmosItemProperties]): Future[Done] = {
     //Each observer is called from a pool managed by CosmosDB ChangeFeedProcessor
     //So its fine to have a blocking wait. If this fails then batch would be reread and
     //retried thus ensuring at-least-once semantics
     val f = eventProducer.send(processDocs(docs, config))
-    Await.result(f, config.feedPublishTimeout)
-    MetricEmitter.emitCounterMetric(feedCounter, docs.size)
-    recordLag(context, docs.last)
+    f.andThen {
+      case Success(_) =>
+        MetricEmitter.emitCounterMetric(feedCounter, docs.size)
+        recordLag(context, docs.last)
+    }
```

Comment:
should we log something on failures here?

Reply:
Logging was being done by CosmosDB SDK for failure but added a log here also now such that we can always capture the failure case